apps/extension/src/ui/domains/Transactions/TxProgress.tsx

```diff
@@ -241,14 +256,21 @@ const TxProgressEvm: FC<TxProgressEvmProps> = ({ tx, className, onClose }) => {
   )
 }
 
-type TxProgressProps = {
+type TxProgressProps<T extends unknown[] = []> = {
   hash: HexString
   networkIdOrHash: string
   onClose?: () => void
   className?: string
+  onSuccess?: (...args: T) => void
 }
 
-export const TxProgress: FC<TxProgressProps> = ({ hash, networkIdOrHash, onClose, className }) => {
+export const TxProgress = <T extends unknown[]>({
+  hash,
+  networkIdOrHash,
+  onClose,
+  className,
+  onSuccess,
```

Comment:
Sorry i didnt dig this thing enough last time. 
It feels like success detection shouldnt originate from here, which is a UI component that is meant to support both EVM and substrate.

Option 1 : use useTransaction(hash) and detect success from useUnbondWizard(), and rollback all changes to this component
Option 2 : move the success check from TxProgressSubstrate up to TxProgress, so it can support both substrate and evm

Big preference for option 1 though

Reply:
Went with option um, thanks for the suggestion!